pkg/yqlib/operator_modulo.go

```diff
@@ -0,0 +1,98 @@
+package yqlib
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	yaml "gopkg.in/yaml.v3"
+)
+
+func createModuloOp(lhs *ExpressionNode, rhs *ExpressionNode) *ExpressionNode {
+	return &ExpressionNode{Operation: &Operation{OperationType: moduloOpType},
+		LHS: lhs,
+		RHS: rhs}
+}
+
+func moduloOperator(d *dataTreeNavigator, context Context, expressionNode *ExpressionNode) (Context, error) {
+	log.Debugf("Modulo operator")
+
+	return crossFunction(d, context.ReadOnlyClone(), expressionNode, modulo, false)
+}
+
+func modulo(d *dataTreeNavigator, context Context, lhs *CandidateNode, rhs *CandidateNode) (*CandidateNode, error) {
+	lhs.Node = unwrapDoc(lhs.Node)
+	rhs.Node = unwrapDoc(rhs.Node)
+
+	lhsNode := lhs.Node
+
+	if lhsNode.Tag == "!!null" {
+		return nil, fmt.Errorf("%v (%v) cannot modulo by %v (%v)", lhsNode.Tag, lhs.GetNicePath(), rhs.Node.Tag, rhs.GetNicePath())
+	}
+
+	target := lhs.CreateReplacement(&yaml.Node{
+		Anchor: lhs.Node.Anchor,
```

Comment:
not sure why this is here? (maybe I need it in the sub/add/mult operators ?

Reply:
Yeah I'm not sure.  I started with the add operator which is where it came from https://github.com/mikefarah/yq/blob/master/pkg/yqlib/operator_add.go#L59-L61

Reply:
I think it isn't necessary because I didn't make `/=` or `%=` operators.  I'll remove it.

Reply:
https://github.com/mikefarah/yq/pull/1593/commits/b499322914d56e49af2e577be9cc9ff9c43dad1c